src/site/content/en/blog/web-otp/index.md

```diff
@@ -188,64 +188,136 @@ The Web OTP API requires a secure origin (HTTPS).
+instance in the [`options`
+object](https://developer.mozilla.org/en-US/docs/Web/API/CredentialsContainer/get#Parameters).
+
+```js
+const signal = new AbortController();
+let timer = setTimeout(() => {
+  signal.abort();
+}, 10 * 1000);
+
+const content = await navigator.credentials.get({
+  otp: { transport:['sms'] },
+  abort: signal
+});
+```
+
+### Use the API declaratively
+
+The code below demonstrates a web component that extends `input`.
+
+```js
+customElements.define("one-time-code",
+  class extends HTMLInputElement {
+    connectedCallback() {
+      this.receive();
+    }
+    async receive() {
+      try {
+        const sms = await navigator.credentials.get({
+          otp: { transport:['sms'] }
+        });
+        this.value = sms.code;
+        this.form.submit();
+      } catch (e) {
+        console.error(e);
+      }
+    }
+  }, {
+    extends: "input"
+});
+```
+After this declaration you can add `is="one-time-code` to any `input` element.
+As soon as the element is added to the document tree, it waits for an SMS
+message to arrive and submits the form as soon as the OTP is passed.
+
+```html
+<form>
+  <input is="one-time-code" autocomplete="one-time-code" required />
+  <input type="submit">
+</form>
 ```
 
-You can now submit the code to the server to verify it.
 
-### Formatting the SMS message
+### Format the SMS message {: #format }
 
 The API itself should look simple enough, but a critical part is to
 format your SMS text message according to a specific convention. The message has
-to be sent after `navigator.sms.receive()` is called and must comply with a
+to be sent after `avigator.credentials.get()` is called and must comply with the
 formatting convention.
 
-The SMS message must be received on the device where `navigator.sms.receive()`
+The SMS message must be received on the device where `avigator.credentials.get()`
 was called.
 
 The message must adhere to the following formatting:
 
 * The origin part of the URL of the website that invoked the API must be
-  preceded by `For: `.
+  preceded by `@`.
 * The URL must contain (for [the time
   being](https://github.com/samuelgoto/sms-receiver/issues/4#issuecomment-528991114))
   a query parameter whose value is the application hash of the user's Chrome
-  instance. (These are static strings. See the table below.)
-* The URL must contain a query parameter `otp` whose value is the OTP.
+  instance. See [Use the application hash](#use-the-hash) for a list of valid hashes.
+* The URL must contain a pound sign ('`#`') followed by the OTP.
+* Optionally, the message may contain additional text for the user.
 
-An example message that can be retrieved by the browser would look like this:
+For example:
 
 ```text
 Your OTP is: 123456.
 
-For: https://sms-receiver-demo.glitch.me/?otp=123456&xFJnfg75+8v
+@https://www.example.com #123456&xFJnfg75+8v
 ```
 
+### Use the application hash {: #use-the-hash }
```

Comment:
We can now get rid of app hash section. It's no longer needed.

Reply:
I almost did, but it's needed until at least when 83 goes to stable. Does the flag in 83 default to on or off?

Reply:
The flag will not be required on 83.

Reply:
I edited it.